pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -331,7 +331,6 @@ public synchronized void removeConsumer(Consumer consumer, boolean isResetCursor
                 // when topic closes: it iterates through concurrent-subscription map to close each subscription. so,
                 // topic.remove again try to access same map which creates deadlock. so, execute it in different thread.
                 topic.getBrokerService().pulsar().getExecutor().submit(() ->{
-                    topic.removeSubscription(subName);
```

Comment:
I think there may be some side effect to remove this line, if `removeConsumer` is not called from `doUnsubscribe`. Maybe it's better to make `topic.removeSubscription` support removing non-exists subName (Just add a null check in `removeSubscription`).